scripts/py/gen_nextmap_maplists.py

```diff
@@ -0,0 +1,179 @@
+                    if numberOfTeams == 7:
+                        if maxPlayersPerStartbox >= 2:
+                            teamffasize_dict['2v2v2v2v2v2v2'].append(mapname)
+                    if numberOfTeams == 8:
+                        if maxPlayersPerStartbox >= 2:
+                            teamffasize_dict['2v2v2v2v2v2v2v2'].append(mapname)
+
+                # if a map didn't have "maxPlayersPerStartbox" set for its startboxes, but it's a teamgame map with startboxes for 2 teams, we'll use playerCount instead:
+                if not "maxPlayersPerStartbox" in contents[i]["startboxesSet"][j].keys() and playerCount and isTeam:
+                    numberOfTeams = 0
+                    for k in contents[i]["startboxesSet"][j]["startboxes"]:
+                        numberOfTeams = numberOfTeams + 1
+                    if numberOfTeams == 2:
+                        x = 0
+                        for l in teamsize_dict:
+                            if x + 2 <= math.floor(playerCount/2) and x + 2 >= math.floor(playerCount/4):
+                                teamsize_dict[l].append(mapname)
+                            x = x + 1
+        
+        # add maps to ffasize_dict
+        if isFfa:
+            x = 0
+            for j in ffasize_dict:
+                if x + 3 <= playerCount and x + 3 >= math.floor(playerCount/2):
+                    ffasize_dict[j].append(mapname)
+                x = x + 1
+                    
+        # add maps to certified and uncertified lists
+        if contents[i]["certified"] == True:
+            certified_maps.append(mapname)
+        else:
+            uncertified_maps.append(mapname)
+        
+        #add maps to 1v1 list
+        if is1v1:
+            maps_1v1.append(mapname)
+
+    combined_dict = {**teamsize_dict,**ffasize_dict,**teamffasize_dict}
+
+    #if combined_dict has empty lists then add ".*" (meaning all maps) to that list
+    for i in combined_dict:
+        if len(combined_dict[i]) == 0:
+            combined_dict[i].append('.*')
+    
+    return combined_dict, certified_maps, uncertified_maps, maps_1v1
+
+def get_output_string(combined_dict, certified_maps, uncertified_maps, maps_1v1):
+    output_string = """# This file was automatically generated by https://github.com/beyond-all-reason/maps-metadata/tree/main/scripts/py/gen_nextmap_maplists.py using data from rowy.
+# Next update from rowy will overwrite this file so do not manually edit this file.
+# If you want to make updates to this see https://github.com/beyond-all-reason/maps-metadata/wiki/Adding-a-created-map-to-the-game.
+# A map needs properly configured playercount, startboxes and maxPlayersPerStartbox in https://rowy.beyondallreason.dev/table/maps to appear here.
+# For example a 2v2v2v2 map needs a startbox configuration with 4 startboxes and maxPlayersPerStartbox >= 2.
+[all]
+.*
+
+[certified]
+"""
+    for i in certified_maps:
+        output_string = output_string + i + '\n'
```

Comment:
instead of doing for loop, and `var = var +`, you can use `str.join` function and `+=`, so:

```py
output_string += '\n'.join(certified_nmaps)
```

Reply:
I see! I changed this now but I think I will make this mistake again and have to learn it a few times before I remember it permanently.